### ISSUE TITLE
All work disappeared after tapping the Save button

### ISSUE THREAD

**Issue by BrackoNe** [IpswichMapper/SwiftAddress#10]
Hi, 

I thought it is on me, but it seems to be an issue in the app.
I have spent an hour collecting all house numbers in one street; once I've clicked the save button, I got a confirmation that data is saved in the SwiftAddress folder.

The folder is missing in the root folder. There is only the old KeyMapper folder.

Now whole work is gone.

![Screenshot_20210218-105822_SwiftAddress](https://user-images.githubusercontent.com/8583745/108346762-0676e400-71e0-11eb-989b-b40638f2d70b.jpg)
![Screenshot_20210218-112025_SwiftAddress](https://user-images.githubusercontent.com/8583745/108346772-08d93e00-71e0-11eb-8568-cbc0dead8557.jpg)

I'm using Samsung Galaxy Note 10 Plus, Android 11.

All permission are allowed.
![Screenshot_20210218-115551_Permission controller](https://user-images.githubusercontent.com/8583745/108347009-52c22400-71e0-11eb-8298-07f3a1b08a30.jpg)

**Comment by IpswichMapper** [comment]
I'm really sorry. This is completely my fault.

 It has been known that there are file saving issues in this app: https://github.com/IpswichMapper/SwiftAddress/issues/6 . The reason for this is because new versions of android (android 11 specifically & android 10) no longer allow creating root directories like the app currently does.

What I should have done is make it clear to not take large surveys using this app. I'm really sorry I didn't clearly state this. I have updated the readme now. I'll also push the next update as quickly as I can to fix this issue.

I have completely reworked the saving mechanism ( https://github.com/IpswichMapper/SwiftAddress/commit/46c987a61db76e48627c0b9b437342aac254def0 ), however this fix is not present in the current update.

**Comment by IpswichMapper** [comment]
Hello, I have made a new release. Hopefully that works:

https://github.com/IpswichMapper/SwiftAddress/releases/tag/v0.2.0

The file saving mechanism has been completely overhauled. Please try adding some fake data and saving, and seeing if you can open the fake data in JOSM

**Comment by BrackoNe** [comment]
Everything works as expected.

![image](https://user-images.githubusercontent.com/8583745/108611219-badd5980-73dc-11eb-816c-e3e0cf85e952.png)

Thank you!

**Comment by IpswichMapper** [comment]
Did you do another survey or is this made up  test data?
-- 



20 Feb 2021, 23:36 by notifications@github.com:

**Comment by BrackoNe** [comment]
Completely new survey. I will have one enhancement proposal, but I will write it in a day or two.

Also one short question: is it really necessary to zip every save? Once it is unpacked, osm file will preserve the same name over and over again.

**Comment by IpswichMapper** [comment]
It is convenient because of images. Each image has to be stored in a separate file, so with the old KeyPad Mapper 3 you would be left with a bunch of image files. If you did multiple surveys, the image files could become confusing.

So in that sense, the zip file is like a "package" that contains all the information relating to the survey you took.

I tried dropping the zip file into JOSM, it attempts to read it and says no data found. I don't know if the native JOSM application is reading it or the "OpenData" plugin. I suspected it is that plugin that is trying to read it.

I could ask the JOSM devs to try and read relevant files (e.g. audio, image. OSM, OSC, GEOJSON) from a zip when you try to drap and drop it into JOSM.

JOSM also has own "session" format that can store osm files, image files, imagery layers and more. I could look into that and see how it works and try and implement that into SwiftAddress. 

As for the .osm file name having the same name, I can add a date-time stamp to the end of the file name if you want. I ddi have this before, but removed it with this zip feature because Android Studio is telling me to avoid the SimpleDateFormat() function even though it works perfectly.
